### PR TITLE
Pcie 208 endpoint dev implement the tlp construction algorithms functions

### PR DESCRIPTION
**Pull request by mEmadEl-Dien**

### CODE REVIEW

File: PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/configuration_algorithm_classes/completion_construction_algorithms.cpp [hunk 1]
```diff
@@ -1,17 +1,42 @@
...
+CompletionConstructionAlgorithms::CompletionConstructionAlgorithms(): dataToBeReadBits(0), registerLengthInBytes(0){}
 
 /**
  * @brief Constructing a Completion TLP with Data Payload inside (return of a Configuration Read request)
  * 
  * @return TLP -> The Constructed TLP (CplD)
  */
+
 TLP CompletionWithData::constructTLP()
 {
```
Comment (david-wagih) [inline PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/configuration_algorithm_classes/completion_construction_algorithms.cpp:12]: Can we extend it to work properly even with memory requests?

Comment (david-wagih) [reply PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/configuration_algorithm_classes/completion_construction_algorithms.cpp:12]: I think we will also need to pass the TLP as a parameter to the function, as I am saying I want a completion for this one specifically.

Comment (mEmadEl-Dien) [reply PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/configuration_algorithm_classes/completion_construction_algorithms.cpp:12]: And for the second comment, indeed I have already made what you are saying and you can find this is the pull request(there is a function called setTLP() inside the TlpConstructor class to do so), it's just un-merged yet.

File: PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/configuration_algorithm_classes/completion_construction_algorithms.cpp [hunk 2]
```diff
@@ -33,8 +71,22 @@ TLP CompletionWithoutData::constructTLP()
  */
 TLP CompletionWithUR::constructTLP()
 {
```
[the remark this answers is not in the document]
Comment (mEmadEl-Dien) [reply PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/configuration_algorithm_classes/completion_construction_algorithms.cpp:73]: The same answer as an above comment.

**Review by david-wagih — APPROVED** [review]
nice work bro 💯

**Pull request merged**